File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        res = 0
        for row in grid:
            l, r = 0, len(row)-1
            while l <= r:
                m = (l+r) // 2
                if row[m] < 0:
                    r = m - 1
                else:
                    l = m + 1
            res += len(row) - l
        return res

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.kt
class Solution {
    fun countNegatives(grid: Array<IntArray>): Int {
        var res = 0
        for (i in grid.indices) {
            val row = grid[i]
            var l = 0
            var r = row.size-1
            while (l <= r) {
                val m = ((l+r)/2)
                if (row[m] < 0) {
                    r = m - 1
                } else {
                    l = m + 1
                }
            }
            res += row.size - l
        }
        return res
    }
}